cmd/snap/cmd_quota.go

```diff
@@ -118,13 +166,62 @@ func parseQuotas(maxMemory string) (*client.QuotaValues, error) {
 		mem = value
 	}
 
+	if cpuMax != "" {
+		countValue, percentageValue, err := parseCpuQuota(cpuMax)
+		if err != nil {
+			return nil, err
+		}
+		if percentageValue > 100 || percentageValue <= 0 {
+			return nil, fmt.Errorf("cpu quota percentage must be between 1 and 100")
+		}
+
+		cpuCount = countValue
+		cpuPercentage = percentageValue
+	}
+
+	if cpuSet != "" {
+		cgv, err := getCGroupVersion()
+		if err != nil {
+			return nil, fmt.Errorf("cgroup v2 is required for --cpu-set: %v", err)
```

Comment:
the error is in determining the version, the code hasn't asserted that it's >=2 yet
```suggestion
			return nil, err
```

Reply:
Fixed! Thanks for pointing it out